minetest/minetest
# [nosquash] Devtest: Add jukebox and branding iron

- The jukebox can be used to play sounds by giving a SimpleSoundSpec, sound parameters and ephemeral. It can also stop and fade the sound. The sound handle is stored in a table per jukebox pos hash, there's a Release button to drop the handle.
- The branding iron can be used to give objects names, i.e. `obj:1`. This is needed to pass objects in the sound parameter table.

Screenshot:
![screenshot_20220917_000803](https://user-images.githubusercontent.com/7613443/190822615-0d006cb3-269d-4106-a107-3d52ce9f0e58.jpg)


## To do

This PR is a Ready for Review.

Don't squash.
Review commits separately.
(A file was renamed.)

## How to test

* Place a jukebox.
* Play around with it.
* To check its usefulness, try for example to play a positional long or looped sound, and fade it with `gain=0, step=1`, and see how it takes 3 seconds, even though the given gain is (by default) 1. Such and other bugs can be tested with this tool.
